src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcBuilderExtensions.cs

```diff
@@ -91,5 +107,27 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
             // Consumed by the Cache tag helper to cache results across the lifetime of the application.
             services.TryAddSingleton<IMemoryCache, MemoryCache>();
         }
+
+        private static IEnumerable<RazorFileInfoCollection> GetFileInfos(IEnumerable<Assembly> assemblies) =>
+            assemblies
+                .SelectMany(assembly => assembly.ExportedTypes)
+                .Where(IsValidRazorFileInfoCollection)
+                .Select(Activator.CreateInstance)
+                .Cast<RazorFileInfoCollection>();
+
+        internal static bool IsValidRazorFileInfoCollection(Type type)
+        {
+            var inAssemblyType = typeof(RazorFileInfoCollection);
+            if (inAssemblyType.IsAssignableFrom(type))
+            {
+                var hasParameterlessConstructor = type.GetConstructor(Type.EmptyTypes) != null;
+
+                return hasParameterlessConstructor
+                    && !type.GetTypeInfo().IsAbstract
+                    && !type.GetTypeInfo().ContainsGenericParameters;
```

Comment:
Should we ignore anything inherited from `RazorFileInfoCollection` without a parameterless constructor? Would it be better to throw because it's a mistake?